src/interactivity/Interactivity.js

```diff
@@ -167,13 +171,39 @@ export default class Interactivity {
         });
     }
 
+    _subscribeToLayerEvents (layers) {
+        layers.forEach(layer => {
+            layer.on('updated', () => {
+                this._onLayerUpdated(layer.isAnimated());
+            });
+        });
+    }
+
     _subscribeToIntegratorEvents (integrator) {
         integrator.on('mousemove', this._onMouseMove.bind(this));
         integrator.on('click', this._onClick.bind(this));
     }
 
+    _onLayerUpdated (isAnimated) {
+        if (isAnimated) {
+            this._updateCalls++;
+            // Debounce calls to fire the events
+            if (this._updateCalls > BATCH_UPDATE_CALLS) {
```

Comment:
I am not sure if we should use a constant here, I am thinking of having the count of steps the animation has.

Reply:
Yes, I'm not very happy with the constant too. Maybe I need to implement a time debounce here.

Reply:
We could something like this:

* If an expression is animated, we should be able to know the animation progress from the expression itself (we will need to check the children recursively). So we can have a `getProgressValue` in the base expression that, if it is animated, calls `getProgressValue` of its children until it gets to the animated value (does it makes sense?)
* Then, in `_onLayerUpdated` method, we could check if the progress has changed in order to trigger the event.

Reply:
I think we need to filter first the `updated` event callbacks and then verify if there is a change in the animation.

There are ~60 calls of `updated` per second that we need to debounce, also we can have more than one animated layer and even more than one animated style property.

Reply:
Check my other comment related to this, pls. Keep in mind that `Animation` is not the only source of "animatedness", so you cannot expect `getProgressValue` to exist somewhere. Moreover, you can mix multiple `Animation`s.

Reply:
I have made a test computing an array of the `getProgressValue` from a viz object

```js
return this._getRootExpressions().filter(expr => expr.getProgressValue).map(expr => expr.getProgressValue());
```

The problem is that it is difficult to change if the progress has changed with multiple style properties animated a multiple layers and it adds significant extra computing time and the point is that the progress always changes for each iteration so I think it can not be used to filter the triggers.

I have tested also the approach of debouncing the `updated` function call with a `max wait` of 100ms, so there will be always 10 events per second if we have one or more animated layers and one or more animated properties in each layer.

The second step is to filter `featureEnter`, `featureLeave` and `featureHover` events. Since we are computing the differences  between featuresEntered and featuresLeft to trigger `featureEnter` and `featureLeave` events, we can use these lists to know if there is any change in the features when the function is called from _onLayerUpdated. So when the mouse moves `featureHover` is always triggered because of coordinates and position change, and when the mouse stays and there is an `updated` debounced event from an animated layer, `featureHover` only triggers if features change.

Reply:
> approach of debouncing the updated function call with a max wait of 100ms, so there will be always 10 events per second

Sorry I think I missed something here and I do not understand this approach very well :( My questions are:
* Why should we debounce the updated function?
* Why 10 events per second?

> we can use these lists to know if there is any change in the features when the function is called from _onLayerUpdated

I like this approach! ✨